### PR TITLE
dotnetCorePackages.sdk_5_0: 5.0.200 -> 5.0.202 and fix build failure of `osu-lazer` and `ryujinx`

### PR DESCRIPTION
**Pull request by oxalica**
###### Motivation for this change
The commit 2026cd27816e06a12371973c167f99d682cdf9f6 bumping `sdk_5_0` to 5.0.200 breaks some packages.
This PR bump it to 5.0.202 which fixes the NuGet issue https://github.com/NuGet/Home/issues/10491.
Fixes build failure of `osu-lazer` and `ryujinx` in current master.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
   8 packages built:
   dotnet-sdk_5 fsharp41 msbuild omnisharp-roslyn opentabletdriver osu-lazer roslyn ryujinx
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119456` at dd299588 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built successfully:</summary>
  <ul>
    <li>dotnet-sdk_5 (dotnetCorePackages.sdk_5_0)</li>
    <li>fsharp41</li>
    <li>msbuild</li>
    <li>omnisharp-roslyn</li>
    <li>opentabletdriver</li>
    <li>osu-lazer</li>
    <li>roslyn</li>
    <li>ryujinx</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/fixup-phase.md">fixup-phase</a>

`fixupPhase` should not be overridden, use `postFixup` instead.

Near pkgs/games/osu-lazer/default.nix:103:3:
```
    |
103 |   fixupPhase = ''
    |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/games/osu-lazer/default.nix:28:15:
```
   |
28 |   patches = [ ./bypass-tamper-detection.patch ];
   |               ^
```
</li>
    <li>warning: unused-argument

Unused argument: `alsaPlugins`.
Near pkgs/games/osu-lazer/default.nix:3:48:
```
  |
3 | , ffmpeg_4, alsaLib, SDL2, lttng-ust, numactl, alsaPlugins
  |                                                ^
```
</li>
  </ul>
</details>

File: pkgs/development/compilers/dotnet/default.nix [hunk 1]
```diff
@@ -124,11 +127,11 @@ rec {
...
   sdk_5_0 = buildNetCoreSdk {
-    version = "5.0.200";
+    version = "5.0.202";
     sha512 = {
-      x86_64-linux = "0g7zcmkcdwc11h42m6hq8d0w55nnvnsmj3dc16829q55cp7l7kggmjljnd9slx7r7nrsyi7yy8brwh8n4kfi5998pdyb09fzhq5w60d";
-      aarch64-linux = "2zy6nxiw313g2sbmnkg76r64llbk2w2wcsa6frq535zbviw52zf163jvw2687rpiw4szdizf337l3b0qa0396abw5dhq2czqlxjyjv8";
-      x86_64-darwin = "2p0yxplafhi5ks38pq8nyi43kpv4l4npa718rvcvl57qs76j0dqlk1s4wdw7msx8g7xxy1aal47zy9rxvlypmgwx4dnp339cmbd6mf6";
+      x86_64-linux = "01ed59f236184987405673d24940d55ce29d830e7dbbc19556fdc03893039e6046712de6f901dc9911047a0dee4fd15319b7e94f8a31df6b981fa35bd93d9838";
+      aarch64-linux = "26ec125a0637e71acad20386474df89a101e9ae948921b5de0cd343f4bc0e84b4e7b2318e15978723eb3b9d321e89b790c3f0424a43c29b0015fc6f2b9e3e9d9";
+      x86_64-darwin = "8f19df4db4146f4749dbf357da9bbc3e2fc5fdee046839b64dae54fbab12623fecea7369e8d1f1b449fb073850f7f1152ecce5fe85e2de52f477f04f6f395d10";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/dotnet/default.nix:134]: Please encode them with base64.

File: pkgs/games/osu-lazer/update.sh [hunk 2]
```diff
@@ -1,8 +1,14 @@
-#!/usr/bin/env nix-shell
-#!nix-shell -i bash -p curl jq common-updater-scripts dotnet-sdk_5
+#!/usr/bin/env bash
```
Comment (SuperSandro2000) [inline pkgs/games/osu-lazer/update.sh:1]: Please revert this.

File: pkgs/misc/emulators/ryujinx/updater.sh [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
 set -eo pipefail
 cd "$(dirname "${BASH_SOURCE[0]}")"
 
+# Requires in-tree dotnet-sdk_5.
+if [[ "$1" != "--in-nix-shell" ]]; then
+    sleep 0.5 # Avoid exec bomb.
+    exec nix-shell -p "with import ../../../.. {}; [ coreutils gnused curl common-updater-scripts nix-prefetch-git jq dotnet-sdk_5 ]" \
+        --run "exec ./updater.sh --in-nix-shell"
+fi
+
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/ryujinx/updater.sh:11]: Please revert this.

**Comment by SuperSandro2000** [comment]
If you need a nix-shell with a new software version please start do nested nix-shells instead of relying on timeouts which can cause fork bombs on slow machines.

**Comment by zyansheep** [comment]
glad to see a PR for this, is there any way I can keep the previous version of osu while updating the rest of my system?

**Comment by oxalica** [comment]
@SuperSandro2000 
When shebang is specified, the specified packages always override outer layer of nix-shell. I tried `-I nixpkgs=../../..` but the path is relative to CWD instead of the script, which is annoying.

> which can cause fork bombs on slow machines.

Never. `sleep` is just a guard for someone making a typo in `if` condition. It isn't necessary.

It's okay to not touch update scripts if it's a better option, but scripts would not able to work until maintainers update their global `nixpkgs` to have this PR included (normally, until this PR gets into nixos-unstable).

**Comment by SuperSandro2000** [comment]
> I reverted changes of update script now, but they would not able to work until maintainers update their global `nixpkgs` to have this PR included (normally, until this PR gets into nixos-unstable).

You should create your PRs from a recent master version anyway and not from nixos-unstable.

**Comment by oxalica** [comment]
@SuperSandro2000 
The issue is that scripts using `nix-shell` as interpreter, cannot easily use current in-tree packages. Even though you checkout master branch, `nix-shell` still resolves `nixpkgs` from NIX_PATH to get packages like `dotnet-sdk_5`, which has different behaviors with the one in master. (Note that running script inside a nix-shell doesn't help, because the inner always overrides the outer)

Currently I:
1. Manually run `nix-shell -p 'with import ./. {}; [ /* ... */ ]'` to import nixpkgs from current directory.
2. Temporarily remove these `nix-shell` shebangs from scripts, or they will override outer nix-shell.
3. Run update script to do a pdate.
4. Revert shebang changes of scripts.

I think this workflow should be automatic when update script depends on in-tree packages.
But I cannot find a better way than `exec` itself in the script.

BTW: The update script is not the main part of this PR. As they are reverted now, I hope this PR can be merged soon.

**Comment by IvarWithoutBones** [comment]
First of all, thanks for fixing the builds and bumping these packages!

> I reverted changes of update script now, but they would not able to work until maintainers update their global nixpkgs to have this PR included (normally, until this PR gets into nixos-unstable).

I don't think there's anything wrong with this, if anyone would like to use the updater scripts in the meantime they can always locally make those changes. I don't think it makes sense to modify the script permanently just  for the few days it takes to hit nixpkgs-unstable, especially since you've already updated the packages them with this PR. It most likely won't be used for the coming few days anyway.

**Review by IvarWithoutBones — APPROVED** [review]
LGTM, thanks for your work!

**Comment by IvarWithoutBones** [comment]
Result of `nixpkgs-review pr 119456` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>dotnet-sdk_5 (dotnetCorePackages.sdk_5_0)</li>
    <li>fsharp41</li>
    <li>msbuild</li>
    <li>omnisharp-roslyn</li>
    <li>opentabletdriver</li>
    <li>osu-lazer</li>
    <li>roslyn</li>
    <li>ryujinx</li>
  </ul>
</details>

**Review by zyansheep — APPROVED** [review]
I am sad to say that my ability to compile and run osu does not reflect on my ability to actually play it :cry:.

**Pull request merged**